src/ohio/baseio.py

```diff
@@ -85,3 +86,41 @@ def readline(self):
                 break
 
         return result
+
+
+class StreamTextIOBase(StreamIOBase, io.TextIOBase):
+    """Readable file-like abstract base class for text.
+
+    Concrete classes may implemented method `__next_chunk__` to return
+    chunks (or all) of the text to be read.
+
+    """
+
+    def __init__(self):
+        self._remainder = ''
+
+    @staticmethod
+    def _get_empty_value():
+        return ''
+
+    @staticmethod
+    def _get_newline():
+        return '\n'
```

Comment:
It seems like there would be utility in keeping it even more condensed than that however.

```
class StreamTextIOBase(io.TextIOBase):

    def __init__(self, buffer_cls=str):
        self._buffer_cls = buffer_cls

        if self._buffer_cls is str:
            self._newline = '\n'
        elif self._buffer_cls is bytes:
            self._newline = b'\n'
        else:
            raise TypeError("StreamTextIOBase supports buffers of type 'str' or 'bytes' not: %r" % self._buffer_cls)

        self._remainder = self._buffer_cls()

    …
```

Not only would that be less boilerplate here, then also perhaps it would be simpler down the line:

```
class IteratorTextIO(baseio.StreamTextIOBase):

    def __init__(self, iterable, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.__iterator__ = iter(iterable)

…
```

And in your bytes iterator case, just: `my_iterator = IteratorTextIO(my_iterable, bytes)`.